pkg/odo/cli/component/create.go

```diff
@@ -942,109 +936,137 @@ func (co *CreateOptions) downloadStarterProject(projectPassed string, interactiv
 		return err
 	}
 
-	var logUrl, url, sparseDir string
-	if project.Git != nil {
-		if strings.Contains(project.Git.Location, "github.com") {
-			url, err = util.GetGitHubZipURL(project.Git.Location, project.Git.Branch, project.Git.StartPoint)
-			if err != nil {
-				return err
-			}
-			sparseDir = project.Git.SparseCheckoutDir
+	log.Info("\nStarter Project")
+
+	if starterProject.Git != nil || starterProject.Github != nil {
+
+		projectSource := common.GitLikeProjectSource{}
+		if starterProject.Git != nil {
+			projectSource = starterProject.Git.GitLikeProjectSource
 		} else {
-			return errors.Errorf("project type git with non github url not supported")
+			projectSource = starterProject.Github.GitLikeProjectSource
+		}
+
+		remoteName, remoteUrl, reference, err := projectSource.GetDefaultSource()
+		if err != nil {
+			return errors.Wrapf(err, "unable to get default project source for starter project %s", starterProject.Name)
+		}
+
+		downloadSpinner := log.Spinnerf("Downloading starter project %s from %s", starterProject.Name, remoteUrl)
+
+		_, err = git.PlainClone(path, false, &git.CloneOptions{
+			URL:           remoteUrl,
+			RemoteName:    remoteName,
+			ReferenceName: plumbing.ReferenceName(reference),
+			SingleBranch:  true,
+			// we don't need history for starter projects
+			Depth: 1,
+		})
+		if err != nil {
+			downloadSpinner.End(false)
+			return err
 		}
-		logUrl = project.Git.Location
-	} else if project.Github != nil {
-		url, err = util.GetGitHubZipURL(project.Github.Location, project.Github.Branch, project.Github.StartPoint)
+
+		// we don't want to download project be a git repo
+		err = os.RemoveAll(filepath.Join(path, ".git"))
+		if err != nil {
+			// we don't need to return (fail) if this happends
+			log.Warning("Unable to delte .git from cloned starter project")
+			downloadSpinner.End(false)
+		}
+
+	} else if starterProject.Zip != nil {
+		url := starterProject.Zip.Location
+		logUrl := starterProject.Zip.Location
+		sparseDir := starterProject.Zip.SparseCheckoutDir
+		downloadSpinner := log.Spinnerf("Downloading starter project %s from %s", starterProject.Name, logUrl)
+		err := checkoutProject(sparseDir, url, path)
 		if err != nil {
+			downloadSpinner.End(false)
 			return err
 		}
-		logUrl = project.Github.Location
-		sparseDir = project.Github.SparseCheckoutDir
-	} else if project.Zip != nil {
-		url = project.Zip.Location
-		logUrl = project.Zip.Location
-		sparseDir = project.Zip.SparseCheckoutDir
+		downloadSpinner.End(true)
```

Comment:
```
[mrinaldas@localhost testdir]$ odo create nodejs --starter
Experimental mode is enabled, use at your own risk

Validation
 ✓  Checking devfile existence [16153ns]
 ✓  Checking devfile compatibility [22241ns]
 ✓  Creating a devfile component from registry: project-update [23088ns]
 ✓  Validating devfile component [125229ns]

Starter Project
 ◓  Downloading starter project nodejs...
Please use `odo push` command to create the component with source deployed
```
the download spinner is not ending properly

Reply:
fixed